libs-haskell/test-utils/DA/Test/Util.hs

```diff
@@ -69,3 +73,14 @@ withEnv vs m = bracket pushEnv popEnv (const m)
                     Nothing -> unsetEnv key
                     Just val -> setEnv key val True
                 pure (key, oldVal)
+
+newtype ShouldSucceed = ShouldSucceed {unShouldSucceed :: Bool}
+
+callProcessSilent :: Maybe ShouldSucceed -> FilePath -> [String] -> IO ()
```

Comment:
Why do we pass in a `Maybe`? There are only two values true and false. Pretending that there is a first value `Nothing` is just confusing.

Reply:
Maybe just make an internal `callProcessSilent'` function that takes the `ShouldSuceeed` argument and then have `callProcessSilent = callProcessSilent' (ShouldSucceed True)` and `callProcessSilentError = callProcessSilent' (ShouldSucceed False)`?